nixos/tests/etcd-cluster.nix

```diff
@@ -0,0 +1,128 @@
+# This test runs simple etcd cluster
+
+import ./make-test.nix ({ pkgs, ... } : let
+  certs = pkgs.runCommand "certs" {
+    buildInputs = [pkgs.openssl];
+  } ''
+    mkdir -p $out
+    openssl genrsa -out $out/ca-key.pem 2048
+    openssl req -x509 -new -nodes -key $out/ca-key.pem -days 10000 -out $out/ca.pem -subj "/CN=etcd-ca"
+
+    cat << EOF > openssl.cnf
+    ions = v3_req
+    distinguished_name = req_distinguished_name
+    [req_distinguished_name]
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment
+    extendedKeyUsage = serverAuth
+    subjectAltName = @alt_names
+    [alt_names]
+    DNS.1 = node1
+    DNS.2 = node2
+    DNS.3 = node3
+    IP.1 = 127.0.0.1
+    EOF
+
+    openssl genrsa -out $out/etcd-key.pem 2048
+    openssl req -new -key $out/etcd-key.pem -out etcd.csr -subj "/CN=etcd" -config openssl.cnf
+    openssl x509 -req -in etcd.csr -CA $out/ca.pem -CAkey $out/ca-key.pem -CAcreateserial -out $out/etcd.pem -days 365 -extensions v3_req -extfile openssl.cnf
+
+    cat << EOF > client-openssl.cnf
+    ions = v3_req
+    distinguished_name = req_distinguished_name
+    [req_distinguished_name]
+    [ v3_req ]
+    basicConstraints = CA:FALSE
+    keyUsage = digitalSignature, keyEncipherment
+    extendedKeyUsage = clientAuth
+    EOF
+
+    openssl genrsa -out $out/etcd-client-key.pem 2048
+    openssl req -new -key $out/etcd-client-key.pem -out etcd-client.csr -subj "/CN=etcd-client" -config client-openssl.cnf
+    openssl x509 -req -in etcd-client.csr -CA $out/ca.pem -CAkey $out/ca-key.pem -CAcreateserial -out $out/etcd-client.pem -days 365 -extensions v3_req -extfile client-openssl.cnf
+  '';
```

Comment:
I've found it helpful to break these out in to separate buildCommand calls so it is clearer what depends on what and how it fits together. Would you be interested in a patch to your PR which breaks each of these in to its own derivation? Does that seem like overkill?